enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java

```diff
@@ -414,30 +418,40 @@ private String lookupUser(final OnboardingProcessEntity process, final Map<Strin
         }
     }
 
-    @SneakyThrows(OnboardingProcessException.class)
-    private OnboardingProcessEntity createNewProcess(final Map<String, Object> identification, final String identificationData) {
-        final OnboardingProcessEntity process = createNewProcess(identificationData);
+    private OnboardingProcessEntity createNewProcess(final Map<String, Object> identification, final String identificationData, final RequestContext requestContext) {
+        final OnboardingProcessEntity process = createNewProcess(identificationData, requestContext);
         logger.debug("Created process ID: {}", process.getId());
         final String userId = lookupUser(process, identification);
         process.setUserId(userId);
         auditService.audit(process, "Process started for user: {}", userId);
         return process;
     }
 
-    private OnboardingProcessEntity createNewProcess(final String identificationData) throws OnboardingProcessException {
+    private OnboardingProcessEntity createNewProcess(final String identificationData, final RequestContext requestContext) {
         final OnboardingProcessEntity process = new OnboardingProcessEntity();
         process.setIdentificationData(identificationData);
         process.setStatus(OnboardingStatus.ACTIVATION_IN_PROGRESS);
         process.setTimestampCreated(new Date());
-        new OnboardingProcessEntityWrapper(process).setLocale(LocaleContextHolder.getLocale());
+        setProcessCustomData(process, requestContext);
         return onboardingProcessRepository.save(process);
     }
 
+    private static void setProcessCustomData(final OnboardingProcessEntity process, final RequestContext requestContext) {
+        final OnboardingProcessEntityWrapper processWrapper = new OnboardingProcessEntityWrapper(process);
+        processWrapper.setLocale(LocaleContextHolder.getLocale());
+        processWrapper.setIpAddress(requestContext.getIpAddress());
+        processWrapper.setUserAgent(requestContext.getUserAgent());
+    }
+
     @SneakyThrows(OnboardingProcessException.class)
-    private OnboardingProcessEntity resumeExistingProcess(final OnboardingProcessEntity process, final Map<String, Object> identification) {
+    private OnboardingProcessEntity resumeExistingProcess(
+            final OnboardingProcessEntity process,
+            final Map<String, Object> identification,
+            final RequestContext requestContext) {
+
         logger.debug("Resuming process ID: {}", process.getId());
         process.setTimestampLastUpdated(new Date());
-        new OnboardingProcessEntityWrapper(process).setLocale(LocaleContextHolder.getLocale());
+        setProcessCustomData(process, requestContext);
```

Comment:
Right now, we are rewriting `ipAddress` and `user-agent` when resuming the process. So the last attempt wins. I can imagine to keep only origin values or both.

Reply:
Yes, this is OK.